include/nektar_interface/particle_cell_mapping/generated_linear/pyramid.hpp

```diff
@@ -0,0 +1,237 @@
+ *
+ * where
+ * d_2 = 1 - xi_2
+ * eta_0 = 2 * ((1 + xi_0) / d_2) - 1
+ * eta_1 = 2 * ((1 + xi_1) / d_2) - 1
+ * eta_2 = xi_2
+ *
+ * # Note vertices C and D are swapped relative to the textbook.
+ * c0 = 0.125 * (1 - eta_0) * (1 - eta_1) * (1 - eta_2)
+ * c1 = 0.125 * (1 + eta_0) * (1 - eta_1) * (1 - eta_2)
+ * c2 = 0.125 * (1 + eta_0) * (1 + eta_1) * (1 - eta_2)
+ * c3 = 0.125 * (1 - eta_0) * (1 + eta_1) * (1 - eta_2)
+ * c4 = (1 + eta_2) * 0.5
+ *
+ *
+ * This is a generated function. To modify this function please edit the script
+ * that generates this function.
+ *
+ * @param[in] xi0 Current xi_n point, x component.
+ * @param[in] xi1 Current xi_n point, y component.
+ * @param[in] xi2 Current xi_n point, z component.
+ * @param[in] v00 Vertex 0, x component.
+ * @param[in] v01 Vertex 0, y component.
+ * @param[in] v02 Vertex 0, z component.
+ * @param[in] v10 Vertex 1, x component.
+ * @param[in] v11 Vertex 1, y component.
+ * @param[in] v12 Vertex 1, z component.
+ * @param[in] v20 Vertex 2, x component.
+ * @param[in] v21 Vertex 2, y component.
+ * @param[in] v22 Vertex 2, z component.
+ * @param[in] v30 Vertex 3, x component.
+ * @param[in] v31 Vertex 3, y component.
+ * @param[in] v32 Vertex 3, z component.
+ * @param[in] v40 Vertex 4, x component.
+ * @param[in] v41 Vertex 4, y component.
+ * @param[in] v42 Vertex 4, z component.
+ * @param[in] phys0 Target point in global space, x component.
+ * @param[in] phys1 Target point in global space, y component.
+ * @param[in] phys2 Target point in global space, z component.
+ * @param[in] f0 Current f evaluation at xi, x component.
+ * @param[in] f1 Current f evaluation at xi, y component.
+ * @param[in] f2 Current f evaluation at xi, z component.
+ * @param[in, out] xin0 Output local coordinate iteration, x component.
+ * @param[in, out] xin1 Output local coordinate iteration, y component.
+ * @param[in, out] xin2 Output local coordinate iteration, z component.
+ */
+inline void newton_step_linear_3d(
+    const REAL xi0, const REAL xi1, const REAL xi2, const REAL v00,
+    const REAL v01, const REAL v02, const REAL v10, const REAL v11,
+    const REAL v12, const REAL v20, const REAL v21, const REAL v22,
+    const REAL v30, const REAL v31, const REAL v32, const REAL v40,
+    const REAL v41, const REAL v42, const REAL phys0, const REAL phys1,
+    const REAL phys2, const REAL f0, const REAL f1, const REAL f2, REAL *xin0,
+    REAL *xin1, REAL *xin2) {
+  const REAL x0 = xi1 + 1;
+  const REAL x1 = xi2 - 1;
+  const REAL x2 = 1.0 / x1;
+  const REAL x3 = x0 * x2;
+  const REAL x4 = x3 + 1;
+  const REAL x5 = v00 * x4;
+  const REAL x6 = xi0 + 1;
+  const REAL x7 = x2 * x6;
+  const REAL x8 = x7 + 1;
+  const REAL x9 = v00 * x8;
+  const REAL x10 = x0 * x6 * 1.0 / (x1 * x1);
+  const REAL x11 = v01 * x4;
+  const REAL x12 = v01 * x8;
+  const REAL x13 = v02 * x4;
+  const REAL x14 = v02 * x8;
+  const REAL J00 =
+      0.5 * v10 * x4 - 0.5 * v20 * x3 + 0.5 * v30 * x0 * x2 - 0.5 * x5;
```

Comment:
ditto 1/2
 + the other files - I guess this is a single change to the python somewhere

Reply:
see https://github.com/ExCALIBUR-NEPTUNE/NESO/pull/186#discussion_r1247717663